base/debian/cuttlefish-base.cuttlefish-host-resources.init

```diff
@@ -136,6 +136,55 @@ destroy_mobile_interface() {
     destroy_tap "${tap}"
 }
 
+# Create a mobile tap interface
+# The tap device will have the IP address and DHCP server running on it, and
+# it will be added to a bridge with no address (for routing).
+# $1 = interface number
+# $2 = ip address base ("a.b.c")
+# $3 = IPv6 prefix ("a:b::")
+# $4 = IPv6 prefix length
+create_wireless_interface() {
+    tap="$(printf cvd-wtap-%02d $1)"
+    gateway="${2}.$((4*$1 - 3))"
+    netmask="/30"
+    network="${2}.$((4*$1 - 4))${netmask}"
+    dhcp_range="${2}.$((4*$1 - 2)),${2}.$((4*$1 - 2))"
+        ipv6_prefix="${3}"
+        ipv6_prefix_length="${4}"
+
+    create_tap "${tap}"
+    ip addr add "${gateway}${netmask}" broadcast + dev "${tap}"
+    if [ -n "${ipv6_prefix}" -a -n "${ipv6_prefix_length}" ]; then
+        ip -6 addr add "${ipv6_prefix}1/${ipv6_prefix_length}" dev "${2}"
+    fi
+    start_dnsmasq \
```

Comment:
Removed dhcp, and checked it's still working well.

Reply:
I believe branches that predate the OpenWRT / mac80211_hwsim virtio code will need the DHCP server to be able to connect to the internet. Can you please test if wifi still works on `android11-gsi` when the DHCP server is removed? Mobile networking has never had DHCP because it gets assigned an IP address in a different way.

Reply:
Sure, I'll test that. By the way, if there're any other build targets worthy to test, please let me know :)

Reply:
Wifi service still works after removing DHCP server. Was DHCP service only for assigning IP for tap devices from the bridge? If so, I think it makes sense.

Reply:
As your concern, it looks wlan0 in CF instance has no assigned IP address unlike mobile network when I tried `adb shell ifconfig`. Let me try it with previous commit.

Reply:
Hmm... same result with applying previous commit only. 

Are there any ways to disable mobile network with `adb shell` for testing wifi? With `android11-gsi`, I cannot access `localhost:8443`, and instances weren't visible in the list when I accessed `localhost:1443`.  So I'm using `adb shell` instead and testing it with `ping google.com`, and I checked it works well. But I'm not sure in 100% the ping comes via wifi, or mobile network.

Reply:
adb root
adb shell settings put global airplane_mode_on 1
adb shell am broadcast -a android.intent.action.AIRPLANE_MODE

For turning on airplane mode to disable mobile network

Reply:
`android11-gsi` runs [VNC by default](https://android.googlesource.com/device/google/cuttlefish/+/android11-gsi/host/commands/assemble_cvd/flags.cc#712) rather than WebRTC, you can either connect to it with a VNC client at port 6444 ( go/cuttlefish-glinux#vnc ) or enable WebRTC explicitly with `launch_cvd --start_webrtc=true`

Reply:
Regardless using dhcp server or not, VirtWifi cannot bring the IP address unlike before. I should figure out what this happens.. Thanks for suggesting testing that!

Reply:
As previous comment, it looks it needs DHCP for legacy wifi feature. It cannot access to the Internet so currently I'm still fixing it, but IP address is assigned for wifi at least when adding DHCP.

Reply:
Fixed! I checked it works for recent `aosp_master` and `android11-gsi` both :)
And dhcp was necessary for the legacy wifi feature before introducing openwrt.